### PR TITLE
Add remaining migrations that we missed in the initial check

### PR DESCRIPTION
**Pull request by dominikschubert**
## Motivation

Migrates older modules to new resource providers but keeps the logic intact as much as possible.

## Changes

- Adds `AWS::Elasticache::Domain`. It's a legacy resource but some might still use it
- Adds `AWS::ResourceGroups::Group`
- Adds `AWS::CloudFormation::Macro`
- Adds `AWS::CloudFormation::WaitCondition`
- Adds `AWS::CloudFormation::WaitConditionHandle`

### CODE REVIEW

File: localstack/services/cloudformation/resource_providers/aws_cloudformation_macro.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    Id: Optional[str]
+    LogGroupName: Optional[str]
+    LogRoleARN: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class CloudFormationMacroProvider(ResourceProvider[CloudFormationMacroProperties]):
+    TYPE = "AWS::CloudFormation::Macro"  # Autogenerated. Don't change
```
Comment (Morijarti) [inline localstack/services/cloudformation/resource_providers/aws_cloudformation_macro.py:30]: Good work on implementing these. I wasn't aware of all the resources that CFn uses. Gonna have to dig in more into documentation

**Review by Morijarti — APPROVED** [review]
Awesome job on getting this done quickly.
LGTM

**Comment by dominikschubert** [comment]
All tests green except one flaky one (`tests/aws/services/stepfunctions/v2/timeouts/test_timeouts.py::TestTimeouts::test_fixed_timeout_service_lambda`)

**Comment by dominikschubert** [comment]
The failing pro integration tests seem to be also caused by a flake (plugin install error for elasticsearch/opensearch)

**Pull request merged**